charts/spire/templates/_spire-lib.tpl

```diff
@@ -248,6 +248,48 @@ to merge in values, but spire needs arrays.
 {{- $config | toPrettyJson }}
 {{- end }}
 
+{{- define "spire-lib.default_securitycontext_values" }}
+allowPrivilegeEscalation: false
+runAsNonRoot: true
+readOnlyRootFilesystem: true
+capabilities:
+  drop: [ALL]
+seccompProfile:
+  type: RuntimeDefault
+{{- end }}
+
+{{- define "spire-lib.securitycontext" }}
+{{ include "spire-lib.securitycontext-extended" (dict "root" . "securityContext" .Values.securityContext) }}
+{{- end }}
+
+{{/* Same as securitycontext but takes in:
+root - global . context for the chart
+securityContext - the subbranch of values that contains the securityContext to merge
+*/}}
+{{- define "spire-lib.securitycontext-extended" }}
+{{- if and (dig "spire" "recommendations" "enabled" false .root.Values.global) (dig "spire" "recommendations" "securityContexts" true .root.Values.global) }}
+{{- $vals := deepCopy (include "spire-lib.default_securitycontext_values" .root | fromYaml) }}
+{{- $vals = mergeOverwrite $vals .securityContext }}
+{{- toYaml $vals }}
+{{- else }}
+{{- toYaml .securityContext }}
+{{- end }}
+{{- end }}
+
+{{/*
+Note: Values not needed on openshift due to it autoassigning restricted users
+*/}}
+{{- define "spire-lib.podsecuritycontext" }}
+{{- $vals := dict "fsGroupChangePolicy" "OnRootMismatch" }}
+{{- if not (dig "openshift" false .Values.global) }}
+{{- $_ := set $vals "runAsUser" 1000 }}
+{{- $_ := set $vals "runAsGroup" 1000 }}
+{{- $_ := set $vals "fsGroup" 1000 }}
```

Comment:
Is it possible to make this yaml like the defult security context section?

Reply:
Yes. But would make for more code. As it would be something like:
```
{{- define "spire-lib.default_podsecuritycontext_values" }}
runAsUser: 1000
runAsGroup: 1000
fsGroup: 1000
{{- end }}
```

Then at 285-286,
```
{{- $vals = mergeOverwrite $vals (include "default_podsecuritycontext_values" .) }}
```

So, 6 lines vs 3 lines.

It does look a little more readable though.

If you like that code better though, I'm happy to do it.

Reply:
Actually, those options are specific to vanilla k8s but don't work on openshift... so maybe:
```
{{- define "spire-lib."default_vanilla_extra_podsecuritycontext_values" }}
```
?

Reply:
Simplicity and readability would be my preference, so contributors don't have to learn to much about expressions when contributing.

Reply:
Ok. I took a stab at reworking this so that there are two more macro's with the defaults for vanilla k8s and openshift. Its denormalized a bit but should be more clear. Please have a look.